pkg/envoy/http_connection_manager.go

```diff
@@ -54,6 +56,17 @@ func NewHTTPConnectionManager(virtualHosts []*route.VirtualHost) httpconnectionm
 				VirtualHosts: virtualHosts,
 			},
 		},
+		GenerateRequestId: &wrappers.BoolValue{
+			Value: true,
+		},
+		// TODO: Read actual tracing information from knative configmaps
+		// HACK: right now, in order to forward the x-b3-* headers properly, we need to enable zipkin tracing
+		// even if it's set to 0% sampling.
```

Comment:
Is this to pass `TestShouldHaveHeadersSet`? The test was actually passed before with Kourier but it started failing after spoof client dropped  x-b3-* headers by https://github.com/knative/pkg/commit/5658d93fb07ee641a4e0155f3f9c262da7199035.

I am trying to add it back in https://github.com/knative/pkg/pull/1447 :thinking:

Reply:
That makes more sense! I will close this one then! thanks @nak3